doc/tutorials/content/grid_filters.rst

```diff
@@ -0,0 +1,69 @@
+.. _grid_filters:
+
+Design of VoxelGrid code structure
+----------------------------------
+
+This document is intended to explain our new ``VoxelGrid`` design and help you implementing your own grid based filters easily by reusing our implementations.
+
+Inheritance structure of VoxelGrid
+----------------------------------
+
+.. image:: images/voxel_grid_code_structure.jpg
+    :height: 150px
+    :align: center 
+
+Structure explanation
+---------------------
+
+We will explain the pieces composing ``VoxelGrid``, then elaborate their purpose and demonstrate how they can be reused to implement your custom grid filters.
+
+In a high level view, ``VoxelGrid`` is comprised from several important classes:
+
+1. ``VoxelFilter``
+2. ``VoxelStruct``
+3. ``Voxel``
+4. ``CartesianFilter``
+5. ``TransformFilter``
+
+
+``VoxelFilter<VoxelStruct<Voxel<PointT>,PointT>>`` is aliased as ``VoxelGrid<PointT>`` in our source code, it implements the API specified for ``VoxelGrid``.
```

Comment:
Why don't we keep things simple and just focus on one class at a time instead of all at once?
* TransformFilter is separate, so it doesn't factor in the tutorial for VoxelGrid at all
* VoxelStruct without template is used to explain the basic needs for VoxelFilter

Thoughts?

Reply:
> the basic needs for VoxelFilter

like requirements of `GridStruct`?

> TransformFilter is separate, so it doesn't factor in the tutorial for VoxelGrid at all

sounds good, I will write a separate tutorial for `TransformFilter`

Reply:
> the basic needs for VoxelFilter

like requirements of `GridStruct`?

> VoxelStruct without template

But would it be better if we introduce also `Voxel` along side? (with template)

> TransformFilter is separate, so it doesn't factor in the tutorial for VoxelGrid at all

sounds good, I will write a separate tutorial for `TransformFilter` .

Reply:
Yes, like requirements of GridStruct, but in a very-basic manner.
> But would it be better if we introduce also Voxel along side?

You can add it, but it's still a tutorial. Make it as approachable for a new comer as possible.

Reply:
I rewrote it yesterday to try to just introduce the `Voxel`, `VoxelStruct` (and its requirement), and `VoxelFilter`, the three main pieces of `VoxelGrid`, while trying to minimize the stuff related to `TransformFilter`. 

Could you take a look to see if it's newcomer-friendly enough? (I don't have much experience in writing such tutorials)